src/sapling/note.h

```diff
@@ -27,17 +27,18 @@ class BaseNote {
 
 class SaplingNote : public BaseNote {
 public:
-    diversifier_t d;
-    uint256 pk_d;
-    uint256 r;
-
-    SaplingNote(diversifier_t d, uint256 pk_d, uint64_t value, uint256 r)
-            : BaseNote(value), d(d), pk_d(pk_d), r(r) {}
+    diversifier_t d = {{0}};
+    uint256 pk_d{UINT256_ZERO};
+    uint256 r{UINT256_ZERO};
 
     SaplingNote() {};
-
+    SaplingNote(diversifier_t d_, uint256 pk_d_, uint64_t value, uint256 r_):
```

Comment:
In other areas of the sources, we are using the underscore at the beginning and not at the end for this scenarios. Would be nice if we follow the same convention here as well.

Reply:
Actually we don't have a uniform standard here really. 
In some places we put the underscore at the beginning, in other places (like in this file elsewhere) after the variable name, yet in other places we append `In` at the end of the variable name, and, finally, in some other places we simply shadow the member name (as it was done here), which should be considered bad practice.

Happy to change it, since I am at it, if we want to start following this convention.

Reply:
Updated.